extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -0,0 +1,122 @@
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+            }
+
+            var content = bindingData.Content.ToObjectFromJson<Dictionary<string, string>>();
+
+            content.TryGetValue("Connection", out var connectionName);
+            content.TryGetValue("BlobName", out var blobName);
+            content.TryGetValue("ContainerName", out var containerName);
+
+            if (string.IsNullOrEmpty(connectionName)
+                || string.IsNullOrEmpty(blobName)
+                || string.IsNullOrEmpty(containerName))
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+            }
+
+            var connectionString = connectionName is null ? null : Environment.GetEnvironmentVariable(connectionName);
+            object result = ToTargetType(context.TargetType, connectionString, containerName, blobName);
+
+            if (result is not null)
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Success(result));
+            }
+
+            return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+        }
+
+        private object? ToTargetType(Type targetType, string connectionString, string containerName, string blobName) => targetType switch
+        {
+            Type _ when targetType == typeof(String)            => GetBlobString(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(Stream)            => GetBlobStream(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(Byte[])            => GetBlobBinaryData(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(BlobClient)        => CreateBlobReference<BlobClient>(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(BlockBlobClient)   => CreateBlobReference<BlockBlobClient>(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(PageBlobClient)    => CreateBlobReference<PageBlobClient>(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(AppendBlobClient)  => CreateBlobReference<AppendBlobClient>(connectionString, containerName, blobName),
+            _ => CreateTargetObject(targetType, connectionString, containerName, blobName)
+        };
+
+        private object CreateTargetObject(Type targetType, string connectionString, string containerName, string blobName)
+        {
+            var content = GetBlobString(connectionString, containerName, blobName);
+            MethodInfo deserializeObjectMethod = GetType()
+                                                .GetMethod(nameof(DeserializeTargetObject), BindingFlags.Instance | BindingFlags.NonPublic)
+                                                .MakeGenericMethod(new Type[] { targetType });
```

Comment:
JsonConvert.DeserializeObject has a non generic overload you can use, so you shouldn't have to do this

Reply:
:o